### PR TITLE
Improvements to commands description

### LINKED ISSUE #293: Forwarding arguments on exec/run/docker escaped by --
This works but has very bad UX and no documentation at all... we might need to think a better way for handling this, really mapping options or improve docs.

### PR DESCRIPTION
**Pull request by fabriciojs**
Closes #293 #19

| Issue | #293 #19 |
| -----: | :-----: |
| :open_book: Documentation | Yes |
| :warning: Break Change | No |

**Description**

Just updating a bunch of descriptions on our commands. This addresses the `--` documentation for most useful commands.

---

**Notes**
- The pattern for options/arguments reference is inspired by `docker` and `kubectl` (while even they fail to have one single solid pattern across the board). This remakes some of the changes on https://github.com/kool-dev/kool/pull/285
- The short description was updated in the same pattern to be indeed a more friendly call of attention for the command and its goal instead of actual usage instruction - reserved to the long description.

Old `kool` output:

```
➜  ~ kool
An easy and robust software development environment
tool helping you from project creation until deployment.
Complete documentation is available at https://kool.dev/docs

Usage:
  kool [command]

Available Commands:
  create      Create a new project using preset
  deploy      Deploys your application using Kool Dev
  docker      Creates a new container and runs the command in it.
  exec        Execute a command within a running service container
  help        Help about any command
  info        Prints out information about kool setup (like environment variables)
  logs        Displays log output from services.
  preset      Initialize kool preset in the current working directory. If no preset argument is specified you will be prompted to pick among the existing options.
  restart     Restart containers - the same as stop followed by start.
  run         Runs a custom command defined at kool.yaml in the working directory or in the kool folder of the user's home directory
  self-update Update kool to latest version
  share       Live share your local environment through an HTTP tunnel with anyone, anywhere.
  start       Start the specified Kool environment containers. If no service is specified, start all.
  status      Shows the status for containers
  stop        Stop and destroy running containers started with 'kool start' command. If no SERVICE is specified as an argument all containers will be stopped.

Flags:
  -h, --help      help for kool
      --verbose   increases output verbosity
  -v, --version   version for kool

Use "kool [command] --help" for more information about a command.
```

And the new `kool` output:

```
➜  ~ kool
An easy and robust software development environment
tool helping you from project creation through deployment.
Making containers adoption easy and affordable. Powering cloud native applications development.

Complete documentation is available at https://kool.dev/docs

Usage:
  kool [command]

Available Commands:
  create      Create a new project using a preset
  deploy      Deploy your local application to Kool Cloud
  docker      Create a new container (a powered up 'docker run')
  exec        Execute a new command inside a running service container
  help        Help about any command
  info        Print out information about your local setup
  logs        Display log output from all or a specific service container
  preset      Install Kool's tailored configuration files in the current directory
  restart     Restart the service containers (the same as 'stop' followed by 'start')
  run         Executes a script defined by kool.yml
  self-update Update kool to the latest version
  share       Live share your local environment to the internet using an HTTP tunnel
  start       Start your service containers from docker-compose.yml
  status      Show the status of all service containers
  stop        Stop and destroy the service containers

Flags:
  -h, --help      help for kool
      --verbose   increases output verbosity
  -v, --version   version for kool

Use "kool [command] --help" for more information about a command.
```

### CODE REVIEW

**Comment by dbpolito** [comment]
I think would be good to add some examples of `--` usage like `--user` or `--network`

**Review by danielsuguimoto — CHANGES_REQUESTED** [review]

File: cmd/docker.go [hunk 1]
```diff
@@ -83,13 +83,13 @@ func (d *KoolDocker) Execute(args []string) (err error) {
 // NewDockerCommand initializes new kool docker command
 func NewDockerCommand(docker *KoolDocker) (cmd *cobra.Command) {
 	cmd = &cobra.Command{
-		Use:   "docker [option...] [image] [command]",
+		Use:   "docker [OPTIONS] IMAGE [COMMAND] -- [ARG...]",
```
[the remark this answers is not in the document]
Comment (fabriciojs) [reply cmd/docker.go:86]: Yeah I dig that! The thing about the special `OPTIONS` situation is that being a so general and common occurrence, the plural seem to save space and be more ease to the eye.

And... I honestly did not make that decision - I took the decision to follow on the predominant pattern I could tell from `docker` and `kubectl` CLI tools themselves.

So I like avoiding having so many `...` and keep OPTIONS in the plural, but I totally get the point on hard semantic standard that tells to have the three dots.

Thoughts @indiegenus @dbpolito ?

**Comment by fabriciojs** [comment]
> I think would be good to add some examples of `--` usage like `--user` or `--network`

@dbpolito I would suggest we handle this in a subsequent task, looking at it more broadly. Cobra package itself allows us for offering examples for the commands, I think we should use that feature to add in savvy examples to key commands we see fit.

**Comment by fabriciojs** [comment]
@indiegenus I have updated the PR as we discussed earlier today. Please, you can now proceed with the language/spelling review and fixes on the updated descriptions. Thanks!

**Pull request merged**